book/modules/creating_modules.md

```diff
@@ -0,0 +1,473 @@
+
+- Commands ([`export def`](/commands/docs/export_def.md))
+- Aliases ([`export alias`](/commands/docs/export_alias.md))
+- Constants ([`export const`](/commands/docs/export_const.md))
+- Known externals ([`export extern`](/commands/docs/export_extern.md))
+- Submodules ([`export module`](/commands/docs/export_module.md))
+- Imported symbols from other modules ([`export use`](/commands/docs/export_use.md))
+- Environment setup ([`export-env`](/commands/docs/export-env.md))
+
+::: tip
+Only definitions marked with `export` (or `export-env` for environment variables) are accessible when the module is imported. Definitions not marked with `export` are only visible from inside the module. In some languages, these would be called "private" or "local" definitions. An example can be found below in [Additional Examples](#local-definitions).
+:::
+
+### `main` Exports
+
+::: important
+An export cannot have the same name as that of the module itself.
+:::
+
+In the [Basic Example](#basic-module-example) above, we had a module named `inc` with a command named `increment`. However, if we rename that file to `increment.nu`, it will fail to import.
+
+```nu
+mv inc.nu increment.nu
+use increment.nu *
+
+# => Error: nu::parser::named_as_module
+# => ...
+# => help: Module increment can't export command named
+# => the same as the module. Either change the module
+# => name, or export `main` command.
+```
+
+As helpfully mentioned in the error message, you can simply rename the export `main`, in which case it will take on the name of the module when imported. Edit the `increment.nu` file:
+
+```nu
+export def main []: int -> int {
+    $in + 1
+}
+```
+
+Now it works as expected:
+
+```nu
+use ./increment.nu
+2024 | inc
+# => 2025
+```
+
+::: note
+`main` can be used for both `export def` and `export extern` definitions.
+:::
+
+::: tip
+`main` definitions are imported in the following cases:
+
+- The entire module is imported with `use <module>` or `use <module.nu>`
+- The `*` glob is used to import all of the modules definitions (e.g., `use <module> *`, etc.)
+- The `main` definition is explicitly imported with `use <module> main`, `use <module> [main]`, etc.)
+
+Conversely, the following forms do _not_ import the `main` definition:
+
+````nu
+use <module> <other_definition>
+# or
+use <module> [ <other_definitions> ]
+:::
+
+::: note
+Additionally, `main` has special behavior if used in a script file, regardless of whether it is exported or not. See the [Scripts](scripts.html#parameterizing-scripts) chapter for more details.
+:::
+
+## Module Files
+
+As mentioned briefly in the Overview above, modules can be created either as:
+
+* `<module_name>.nu`: Useful for simple modules
+* `<module_name>/mod.nu`: Useful for organizing larger module projects where submodules can easily map to subdirectories of the main module
+
+The `increment.nu` example above is clearly an example of the former. Let's convert it to the directory form:
```

Comment:
Makes sense - Thanks!  I also added the terms "file form" and "directory form" directly to the bullets, and converted the bullets to an ordered list so that I can refer to it more explicitly.